### PR TITLE
FEATURE: QA-18566 review and update README in sdk-ui-tests-e2e

### PR DESCRIPTION
**Pull request by TienBien**
---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13205/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14904/ : FAILURE in 38s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13231/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14934/ : FAILURE in 18s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13232/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14935/ : FAILURE in 19s

File: libs/sdk-ui-tests-e2e/README.md [hunk 1]
```diff
@@ -6,37 +6,58 @@ This project contains End-to-End tests for GoodData.UI components.
...
+export TEST_BACKEND=https://abc.your-domain.com
+export TEST_BACKEND_NO_PREFIX=abc.your-domain.com
+export HOST=$TEST_BACKEND
+export IMAGE_ID=ui-sdk-scenarios
+
+```
+
+Provide all necessary info in the `.env` file. You can find more information in the file `.env.template`.
+
+For Tiger
```
Comment (huyenthanh09) [inline libs/sdk-ui-tests-e2e/README.md:23]: should mention TIGER_DATASOURCES_NAME for tiger backend. e.g: TIGER only - Set TIGER_DATASOURCES_NAME to specify data source to provide data for workspace

File: libs/sdk-ui-tests-e2e/README.md [hunk 2]
```diff
@@ -6,37 +6,58 @@ This project contains End-to-End tests for GoodData.UI components.
...
+## DEV Guide for isolated tests
+
+### Running tests locally
 
-The test application runs on `http://localhost:9500/gooddata-ui-sdk?scenario=<scenario-key>`, to show the Scenario you want to test, replace the `<scenario-key>` variable
-with the value of respective Scenario key. You can find all available keys in the `ComponentResolver.tsx`.
+To run the tests against a mock backend follow these steps:
 
-To log in to the scenario app, visit `http://localhost:9500/gdc/account/login`
+-   Run `rush build -t sdk-ui-tests-e2e` to have the latest changes applied to the E2E tests package.
```
Comment (huyenthanh09) [inline libs/sdk-ui-tests-e2e/README.md:49]: it seems like that there is a command to run isolated locally in package.json. Please try to this command if it works **run-isolated-local**, then we should replace these lines by this command

File: libs/sdk-ui-tests-e2e/README.md [hunk 3]
```diff
@@ -6,37 +6,58 @@ This project contains End-to-End tests for GoodData.UI components.
...
 -   In the `.env` file, type the spec file name into `FILTER` variable, file name must be fully specified (e.g. dateFilter.spec.ts). This will keep all the recordings untouched and creates the recording for the new spec only.
--   Run command `npm run run-isolated-record` to create new recordings for tests.
-
-### Test results
-
--   If tests fail, there are 3 ways how to check what happened:
-    -   screenshots in `cypress/screenshots`
-    -   videos in `cypress/videos`
-    -   if you are running tests in record mode, then you can see all calls logged in `recording/mappings`
+-   Run command `rushx run-isolated-record` to create new recordings for tests.
```
Comment (huyenthanh09) [inline libs/sdk-ui-tests-e2e/README.md:60]: please keep "Test results" part here

File: libs/sdk-ui-tests-e2e/README.md [hunk 4]
```diff
@@ -45,16 +66,16 @@ Currently, our ECR images do not support ARM64 tags, so to run Docker things in
...
-### Run tests against wiremock backend with live Cypress controls (tests are importing their recorded mapping files)
+### Run tests against mock backend with live Cypress controls (tests are importing their recorded mapping files)
 
 Good for debugging the tooling when tests are passing with live backend but fail on CI.
 Runs Cypress, Scenarios, and Wiremock separately in Docker.
 Make sure you have set correct SDK_BACKEND environment variable (.env)
 
--   Start recorded backend `npm run local-dev-wiremock` (add `--verbose` param for debugging)
--   Start Scenarios `npm run run-scenarios-local`
+-   Run `rushx create-ref-workspace` to create reference workspace or make sure you have valid TEST_WORKSPACE_ID specified in the `.env` file.
```
Comment (huyenthanh09) [inline libs/sdk-ui-tests-e2e/README.md:75]: should back to old contend and check that if it work. Your new change is mentioning "run isolated record" without live cypress control

File: libs/sdk-ui-tests-e2e/README.md (hunk 1 above)
Comment (TienBien) [reply libs/sdk-ui-tests-e2e/README.md:23]: I think the user can found that variable in .evn.template as I mentioned in line 21.

File: libs/sdk-ui-tests-e2e/README.md [hunk 5]
```diff
@@ -135,7 +124,7 @@ rush build -t sdk-ui-tests-e2e && yarn build-scenarios && \
 To debug tests in visual mode, you need to first build the scenarios and start them in docker compose
 
 ```
-rush build -t sdk-ui-tests-e2e && yarn build-scenarios && \
+rush build -t sdk-ui-tests-e2e && rushx build-scenarios && \
```
Comment (huyenthanh09) [inline libs/sdk-ui-tests-e2e/README.md:127]: the same previous command

File: libs/sdk-ui-tests-e2e/README.md [hunk 6]
```diff
@@ -6,37 +6,58 @@ This project contains End-to-End tests for GoodData.UI components.
 
 Tests in this library verify various go-through scenarios for GoodData.UI components. Tests are implemented using [Cypress](https://www.cypress.io/).
 There are two categories of tests: isolated and integrated. Isolated tests run against recorded backend responses, whereas integrated tests
-run against a live backend. Isolated tests run in the pre-merge phase and itegrated tests run in post-merge phase.
+run against a live backend. Isolated tests run in the pre-merge phase and integrated tests run in post-merge phase.
 
-## DEV Guide for isolated tests
+## Prepare env variables
```
Comment (huyenthanh09) [inline libs/sdk-ui-tests-e2e/README.md:11]: below this part, we should write a part to say create a reference work space and how to update the workspace fixture use **export-fixture** command. Can copy Reference workspace part from dashboard/analytical

File: libs/sdk-ui-tests-e2e/README.md (hunk 2 above)
Comment (TienBien) [reply libs/sdk-ui-tests-e2e/README.md:49]: it just rebuild rush for sdk-ui-tests-e2e, it is redundant so I will remove it.

Comment (huyenthanh09) [reply libs/sdk-ui-tests-e2e/README.md]: Please try to this command if it works run-isolated-local, then we should replace these lines by this command

File: libs/sdk-ui-tests-e2e/README.md [hunk 7]
```diff
@@ -92,40 +113,8 @@ export const ExampleScenario: React.FC = () => {
...
--   PASSWORD=
--   AUTH_TOKEN=
-
-Create reference workspace with `yarn create-ref-workspace` or make sure you have valid TEST_WORKSPACE_ID specified in the `.env` file.
-
-To run the tests locally, you would need to build the sdk and the scenarios against which the tests run and then run the tests, i.e.
-
 ```
-rush build -t sdk-ui-tests-e2e && yarn build-scenarios && \
+rush build -t sdk-ui-tests-e2e && rushx build-scenarios && \
```
[the remark this answers is not in the document]
Comment (huyenthanh09) [reply libs/sdk-ui-tests-e2e/README.md:136]: please check this

File: libs/sdk-ui-tests-e2e/README.md (hunk 5 above)
Comment (huyenthanh09) [reply libs/sdk-ui-tests-e2e/README.md:146]: please check this

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13319/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14995/ : SUCCESS in 8m 28s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13323/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14999/ : SUCCESS in 7m 42s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13324/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15001/ : SUCCESS in 7m 43s

**Review by kandl — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3719

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2693/ : SUCCESS in 11m 03s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/574/ : FAILURE in 1m 15s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/15043/ : SUCCESS in 7m 43s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/192/ : SUCCESS in 8m 04s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/224/ : SUCCESS in 12m 01s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3848/ : SUCCESS in 13s

**Pull request merged**